test/Control/Effect/Spec.hs

```diff
@@ -1,122 +0,0 @@
-{-# LANGUAGE FlexibleContexts, FlexibleInstances, GeneralizedNewtypeDeriving, MultiParamTypeClasses, MultiWayIf, TemplateHaskell, TypeApplications, TypeOperators, UndecidableInstances #-}
-{-# OPTIONS_GHC -O2 -fplugin Test.Inspection.Plugin #-}
-module Control.Effect.Spec
-( spec
-) where
-
-import Control.Carrier
-import Control.Carrier.Error.Either
-import Control.Carrier.Fail.Either
-import Control.Carrier.Reader
-import Control.Carrier.State.Strict
-import Control.Effect.Sum
-import Prelude hiding (fail)
-import Test.Hspec
-import Test.Inspection as Inspection
-
-spec :: Spec
-spec = do
-  inference
-  reinterpretation
-  interposition
-  fusion
-
-inference :: Spec
-inference = describe "inference" $ do
-  it "can be wrapped for better type inference" $
-    run (runHasEnv (runEnv "i" ((++) <$> askEnv <*> askEnv))) `shouldBe` "ii"
-
-askEnv :: Has (Reader env) sig m => HasEnv env m env
-askEnv = ask
-
-runEnv :: env -> HasEnv env (ReaderC env m) a -> HasEnv env m a
-runEnv r = HasEnv . runReader r . runHasEnv
-
-
-newtype HasEnv env m a = HasEnv { runHasEnv :: m a }
-  deriving (Applicative, Functor, Monad)
-
-instance Carrier sig m => Carrier sig (HasEnv env m) where
-  eff = HasEnv . eff . handleCoercible
-
-
-reinterpretation :: Spec
-reinterpretation = describe "reinterpretation" $ do
-  it "can reinterpret effects into other effects" $
-    run (runState "a" ((++) <$> reinterpretReader (local ('b':) ask) <*> get)) `shouldBe` ("a", "baa")
-
-reinterpretReader :: ReinterpretReaderC r m a -> StateC r m a
-reinterpretReader = runReinterpretReaderC
-
-newtype ReinterpretReaderC r m a = ReinterpretReaderC { runReinterpretReaderC :: StateC r m a }
-  deriving (Applicative, Functor, Monad, MonadFail)
-
-instance (Carrier sig m, Effect sig) => Carrier (Reader r :+: sig) (ReinterpretReaderC r m) where
-  eff (L (Ask       k)) = ReinterpretReaderC get >>= k
-  eff (L (Local f m k)) = do
-    a <- ReinterpretReaderC get
-    ReinterpretReaderC (put (f a))
-    v <- m
-    ReinterpretReaderC (put a)
-    k v
-  eff (R other)         = ReinterpretReaderC (eff (R (handleCoercible other)))
-
-
-interposition :: Spec
-interposition = describe "interposition" $ do
```

Comment:
1. This exists in the library, so reimplementing it is particularly purposeless.

2. It’s going away in #223 anyhow.

3. It’s also the wrong way to do this.

4. Plus there’s no motivation, it’s a bad example, etc.

🔥